src/libstd/sys/unix/fs.rs

```diff
@@ -459,9 +459,18 @@ pub fn rmdir(p: &Path) -> io::Result<()> {
 }
 
 pub fn readlink(p: &Path) -> io::Result<PathBuf> {
+    #[cfg(not(target_libc = "newlib"))]
```

Comment:
I understand, and I would except the PNaCl cross currently requires use of my LLVM (which is a merge of `pnacl-llvm` && `rust-llvm`) to work. I'm in the process of configuring Rust to use `pnacl-llvm` from the PNaCl toolchain when PNaCl is targeted, but that part is still a work in progress (thankfully, the NaCl team has committed to keeping their LLVM closer to trunk, so should remain usable should you guys decide to move your LLVM forward). Also, due to the LLVM debacle, this cross needs the `nacl-binutils` headers so LLVM can build `libLTO.so` and `LLVMgold.so` for use during linking && archiving (this part will be removed once `pnacl-llvm` is used).

I initially split this up to make it easier for reviewing, though maybe I was a bit overzealous. :smiley: I can definitely combine all the crate changes into one, large, PR.